cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -50,15 +56,33 @@ FileObject<LogCollection> LogFileManager::readBatch(
     Aws::String aws_line(line.c_str());
     Aws::Utils::Json::JsonValue value(aws_line);
     Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
-    actual_batch_size++;
-    log_set.insert(input_event);
-    data_tokens.push_back(data_token);
+    pq.push(std::make_tuple(input_event.GetTimestamp(), line, data_token));
+  }
+  
+  long latestTime = std::get<0>(pq.top());
+  LogCollection log_data;
+  std::list<FileManagement::DataToken> data_tokens;
+  size_t actual_batch_size = 0;
+  while(!pq.empty()){
```

Comment:
can you comment on the logic here? If i am reading it correctly, it's something like

```
// Pull up to `batch_size` elements from the file manager. Sort by timestamp, throwing away any entries more than 24 hours old, they will not be uploaded. We may return fewer than `batch_size` elements - if there had been more available from the file manager, they would have been older than 24h anyway
```

If that's not correct, should we continue pulling until we find `batch_size` "new-enough" elements?

Reply:
Pinging this again because it hasn't been addressed - it applies to above comment block